File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    var maxsum=Number.MIN_SAFE_INTEGER;
    var csum=0;
    var i=0;
    while(i<nums.length){
        csum+=nums[i];
        maxsum=Math.max(maxsum,csum);
        if(csum<=0){
            csum=0;
        }
        i++;
    }
    return maxsum;
};

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxsum=INT_MIN;
        int csum=0;
        int i=0;
        while(i<nums.size()){
            
            csum+=nums[i];
            maxsum=max(csum,maxsum);
            if(csum<=0){
                csum=0;
            }
            i++;
        }
        return maxsum;
    }
};